### PR TITLE
[SPARK-13746][Tests]stop using deprecated SynchronizedSet

### PR DESCRIPTION
**Pull request by wilswu**
trait SynchronizedSet in package mutable is deprecated

### CODE REVIEW

File: core/src/test/scala/org/apache/spark/ContextCleanerSuite.scala [hunk 1]
```diff
@@ -586,10 +586,12 @@ class CleanerTester(
   }
 
   private def isAllCleanedUp =
-    toBeCleanedRDDIds.isEmpty &&
-    toBeCleanedShuffleIds.isEmpty &&
-    toBeCleanedBroadcstIds.isEmpty &&
-    toBeCheckpointIds.isEmpty
+    synchronized {
```
Comment (srowen) [inline core/src/test/scala/org/apache/spark/ContextCleanerSuite.scala]: This isn't synchronizing access to these sets on the same lock.

Comment (wilson888888888) [inline core/src/test/scala/org/apache/spark/ContextCleanerSuite.scala]: @srowen 
Fixed.  Thanks!

File: core/src/test/scala/org/apache/spark/ContextCleanerSuite.scala [hunk 2]
```diff
@@ -442,25 +442,25 @@ class CleanerTester(
     checkpointIds: Seq[Long] = Seq.empty)
   extends Logging {
 
-  val toBeCleanedRDDIds = new HashSet[Int] with SynchronizedSet[Int] ++= rddIds
-  val toBeCleanedShuffleIds = new HashSet[Int] with SynchronizedSet[Int] ++= shuffleIds
-  val toBeCleanedBroadcstIds = new HashSet[Long] with SynchronizedSet[Long] ++= broadcastIds
-  val toBeCheckpointIds = new HashSet[Long] with SynchronizedSet[Long] ++= checkpointIds
+  val toBeCleanedRDDIds = new HashSet[Int] ++= rddIds
```
Comment (srowen) [inline core/src/test/scala/org/apache/spark/ContextCleanerSuite.scala]: Can this just be `= HashSet(rddIds)` and similarly for the next 4?

File: core/src/test/scala/org/apache/spark/ContextCleanerSuite.scala [hunk 3]
```diff
@@ -579,17 +579,20 @@ class CleanerTester(
 
   private def uncleanedResourcesToString = {
     s"""
-      |\tRDDs = ${toBeCleanedRDDIds.toSeq.sorted.mkString("[", ", ", "]")}
-      |\tShuffles = ${toBeCleanedShuffleIds.toSeq.sorted.mkString("[", ", ", "]")}
-      |\tBroadcasts = ${toBeCleanedBroadcstIds.toSeq.sorted.mkString("[", ", ", "]")}
+       |\tRDDs = ${toBeCleanedRDDIds.synchronized
```
Comment (srowen) [inline core/src/test/scala/org/apache/spark/ContextCleanerSuite.scala]: At this point it's getting hard to read this in the interpolated string. I'd just pull out the calls to `mkString` and save them as `val`s.

**Comment by srowen** [comment]
This needs a rebase anyway so I left a few more comments. Otherwise LGTM pending tests

**Comment by andrewor14** [comment]
ok to test

File: core/src/test/scala/org/apache/spark/ContextCleanerSuite.scala (hunk 2 above)
Comment (wilson888888888) [inline core/src/test/scala/org/apache/spark/ContextCleanerSuite.scala]: @srowen 
I tried to change to val toBeCleanedRDDIds = HashSet(rddIds), but  I got error at  
toBeCleanedRDDIds -= rddId
so I will keep val toBeCleanedRDDIds = new HashSet[Int] ++= rddIds

File: core/src/test/scala/org/apache/spark/ContextCleanerSuite.scala (hunk 3 above)
Comment (wilson888888888) [inline core/src/test/scala/org/apache/spark/ContextCleanerSuite.scala]: @srowen 
I will change this.  Thanks!

File: core/src/test/scala/org/apache/spark/ContextCleanerSuite.scala (hunk 2 above)
Comment (srowen) [inline core/src/test/scala/org/apache/spark/ContextCleanerSuite.scala]: Ah right Scala doesn't have a constructor that builds a set from a list/seq, so this makes a set of a `Seq[Int]`. Yes this is fine.

File: core/src/test/scala/org/apache/spark/ContextCleanerSuite.scala [hunk 4]
```diff
@@ -578,18 +578,27 @@ class CleanerTester(
   }
 
   private def uncleanedResourcesToString = {
+    val s1 = {toBeCleanedRDDIds.synchronized {
```
Comment (srowen) [inline core/src/test/scala/org/apache/spark/ContextCleanerSuite.scala]: These two statements have extra braces

Comment (wilson888888888) [inline core/src/test/scala/org/apache/spark/ContextCleanerSuite.scala]: @srowen 
Sorry, I overlooked this.  Will change now.  Thanks!

**Comment by srowen** [comment]
Merged to master

**Pull request closed without merging**